src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -752,7 +780,17 @@ public FormValidation doCheckExtraHostsString(@QueryParameter String extraHostsS
             final List<String> extraHosts = splitAndFilterEmptyList(extraHostsString, "\n");
             for (String extraHost : extraHosts) {
                 if (extraHost.trim().split(":").length < 2) {
-                    return FormValidation.error("Wrong extraHost {}", extraHost);
+                    return FormValidation.error("Wrong extraHost format: '%s'", extraHost);
+                }
+            }
+            return FormValidation.ok();
+        }
+
+        public FormValidation doCheckSecurityOptsString(@QueryParameter String securityOptsString) {
+            final List<String> securityOpts = splitAndFilterEmptyList(securityOptsString, "\n");
+            for (String securityOpt : securityOpts) {
+                if (securityOpt.trim().split("=").length < 2) {
+                    return FormValidation.error("Wrong security option format: '%s'", securityOpt);
```

Comment:
According to https://docs.docker.com/engine/reference/run/#security-configuration, while _most_ of the `--security-opt` arguments are of the form "name=value", there's at least one that isn't:
> --security-opt="no-new-privileges:true|false"   : Disable/enable container processes from gaining new privileges

...and this is then further confused by the example:
> docker run --security-opt no-new-privileges -it centos bash

which doesn't have a `:true` or `:false` at the end.

As-is, this code would complain to the user if they specified that option in either form.

Reply:
The documentation does not state whether there could be more of these options. I could either add a value checker for `key:label` values in general or specifically check for `no-new-privileges`. I think that last option is safer towards the current documentation which refers to a kernel documentation. 

Which would you prefer?

Reply:
Well, we don't want this plugin code to have to be updated every time docker gets enhanced, so we want to decouple our code from the docker implementation as much as possible...

Suggestion: if there's something that isn't "name=value" and isn't `no-new-privileges`, `no-new-privileges:true` or `no-new-privileges:false` then we raise a warning (not an error) indicating that this "does not look like a valid security option".

My thinking here is that:
 - as we can't be 100% certain that it _can't_ be valid then we shouldn't indicate that it's an error, which is why we use a warning.
 - it looks like options are _usually_ of the form "name=value" which suggests (to me) that any new options added in the future are likely to be of this form too.
 - we shouldn't state as fact anything we can't be 100% sure of, which is why we say "may be invalid" rather than "is invalid".

FYI: If you would like the UI to include localized strings for these messages (e.g. so you can have the error in German when the browser locale says that's preferable) then you can do that - you'd add a line (e.g. `DockerTemplateBase.unrecognizedSecurityOpt={0} may not be a valid security option.`) to `src/main/resources/com/nirima/jenkins/plugins/docker/Messages*.properties`, which would then create a new pair of methods in `com.nirima.jenkins.plugins.docker.Messages`, which you'd then call from the validator.  However, personally, I'd suggest that adding localization support is a much bigger, and separate, issue that'd be best tackled as a separate PR.